src/assets/htmx.md

```diff
@@ -0,0 +1,347 @@
+---
+title: htmx
+authorName: Niraj Acharya
+authorAvatar: https://avatars.githubusercontent.com/u/122071597?v=4
+authorLink: https://github.com/nirajacharya2
+createdAt: Sep 29, 2023
+tags: js, html, go, htmx
+banner: https://raw.githubusercontent.com/JankariTech/blog/master/src/imgs/fallback_banner.png
+---
+
+Htmx is ui library written in JavaScript that extends the features of html. It can be extended to make AJAX requests, do CSS transitions, connect to WebSockets and handle Server Sent Events inside a html document with just some delacritive html attributes.
+
+Htmx was created by Carson Gross in 2013 because he found html to be limited for what he was trying to achieve. Those limitations of html htmx tries to solve are:
+
+- only `<a>` and `<form>` tags are able to make HTTP requests
+- only `click` & `submit` events trigger them
+- only `GET` & `POST` methods are available
+- and when using `<a>` and `<form>` the whole screens gets replaced
+
+In this blog, we will learn how htmx enhances html and overcomes the limitations mentioned above.
+
+## Installing htmx
+
+We can simply install htmx by adding a script tag in our html.
+
+```html
+<head>
+  <script src="https://unpkg.com/htmx.org@1.9.5"></script>
+</head>
+```
+
+Htmx claims to be small (~14k min.gz’d), dependency-free, extendable, IE11 compatible & 67% less code when compared with react
+
+Now let's see how to make a simple ajax request in htmx
+
+## making AJAX request (GET)
+
+First, let's make an endpoint to get data from and i will be using go for this
+
+```go
+func getRoot(w http.ResponseWriter, r *http.Request) {
+    http.ServeFile(w, r, "./static/index.html")
+}
+
+func hi(w http.ResponseWriter, r *http.Request) {
+    io.WriteString(w, "<div>hi</div>")
+}
+func main() {
+
+    http.HandleFunc("/", getRoot)
+    http.HandleFunc("/hello", hi)
+    http.ListenAndServe(":3333", nil)
+}
+```
+
+In the above code example, we made 2 endpoints. `/` to serve the html file and `/hello` will return a html element. all htmx does is get the body of the response and render it on the page so we have to return html in the response.
```

Comment:
```suggestion
In the above code example, we made 2 endpoints. `/` to serve the html file and `/hello` will return a html element. htmx gets the body of the response and renders it on the page, therefore the response of our function should be some html code.
```

I'm not sure if I understand the last part of the sentence properly. Do you want to say the the output needs to be an HTML code snippet?